exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java

```diff
@@ -20,63 +20,49 @@
 import static org.junit.Assert.assertTrue;
 
 import com.google.common.collect.ImmutableList;
-import java.io.BufferedReader;
-import java.io.InputStreamReader;
-import java.net.ServerSocket;
+import java.io.IOException;
 import java.util.ArrayList;
-import org.junit.After;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.builder.ImageFromDockerfile;
 
 @RunWith(JUnit4.class)
 public class EndToEndTest {
-
-  private static final String LOCALHOST = "127.0.0.1";
-
-  private Process mockServerProcess;
   private MetricExporter exporter;
   private MockCloudMetricClient mockClient;
 
-  @Before
-  public void setup() throws MockServerStartupException {
-    try {
-      // Find a free port to spin up our server at.
-      ServerSocket socket = new ServerSocket(0);
-      int port = socket.getLocalPort();
-      String address = String.format("%s:%d", LOCALHOST, port);
-      socket.close();
-
-      // Start the mock server. This assumes the binary is present and in $PATH.
-      // Typically, the CI will be the one that curls the binary and adds it to $PATH.
-      String[] cmdArray =
-          new String[] {System.getProperty("mock.server.path"), "-address", address};
-      ProcessBuilder pb = new ProcessBuilder(cmdArray);
-      pb.redirectErrorStream(true);
-      mockServerProcess = pb.start();
-
-      // Setup the mock metric client
-      mockClient = new MockCloudMetricClient(LOCALHOST, port);
+  /** A test-container instance that loads the Cloud-Ops-Mock server container. */
+  private static class CloudOperationsMockContainer
+      extends GenericContainer<CloudOperationsMockContainer> {
+    CloudOperationsMockContainer() {
+      super(
+          new ImageFromDockerfile()
+              .withDockerfileFromBuilder(
+                  builder ->
+                      builder
+                          .from("golang")
+                          .run("go get github.com/googleinterns/cloud-operations-api-mock/cmd")
```

Comment:
I think the version of the mock API should be pinned somewhere

Reply:
Agreed.  I'm happy to pin to a release, but which one?  *is* there a release that corresponds to the current binary?

Reply:
Ya the releases and tags in that repo should be correct https://github.com/googleinterns/cloud-operations-api-mock/releases. The script had `v2-alpha` in there, hopefully go accepts that

Reply:
It doesn't.  The tags have to be per-module. I think @punya could likely go tag some stuff for us quickly OR I could suck it up and learn me some go go gadget modules.